src/FSharpx.Collections/NonEmptyList.fs

```diff
@@ -5,29 +5,27 @@ open System.Collections
 open System.Collections.Generic
 open System.Runtime.CompilerServices
 open FSharpx
-
+ 
 type NonEmptyList<'T> = 
-    { Head: 'T
-      Tail: 'T list }
+    private { List: 'T list }
```

Comment:
Can do, but this would require implementing (and testing) an awful lot of things we get for free with a record type, such as structural equality, comparison, etc.